cli_controller.py

```diff
@@ -0,0 +1,138 @@
+        self.load_tree()
+
+
+    """ 
+    Loads all files into command structure
+    """
+    def load_tree(self):
+        for file in self.command_file_list:
+            self.load_commands(file)
+
+
+    """
+    Loads a single JSON file into command structure
+    """
+    def load_commands(self, file):
+        with open(file, "r") as json_file:
+            data = json.load(json_file)
+            try:
+
+                # Check how far the path already exists
+                # - If the command has some of the preserved keywords, the program exits.
+                # - Any missing links in the tree will be added
+
+                for command in data["commands"]:
+                    for command_type in self.global_commands:
+                        if command["target"].upper() in self.global_commands[command_type]:
+                            exit("Used keyword {} as target. Using keywords is prohibited!".format(command["target"]))
+                    current_node = self.root_node
+
+                    # Per path checking if it has a child
+
+                    for path_piece in command["path"].split(" "):
+                        for command_type in self.global_commands:
+                            if path_piece.upper() in self.global_commands[ command_type]:
+                                exit("Used keyword {} as target. Using keywords is prohibited!".format(command["target"]))
+
+                        # If the current path info already exists, traverse the tree
+                        # Else, add the missing link
+
+                        if current_node.has_child(path_piece):
+                            current_node = current_node.get_child(path_piece)
+                        else:
+                            new_node = CommandNode(path_piece, current_node)
+                            new_node.set_parent(current_node)
+                            current_node.add_child(new_node)
+                            current_node = new_node
+
+                    # After all the missing links in the tree are made, add the command
+                    target_node = CommandNode(
+                        command["target"],
+                        parameter_list = command["parameters"],
+                        command_info = command["info"]
+                    )
+                    target_node.set_parent(current_node)
+                    current_node.add_child(target_node)
+            except KeyError as error:
+                print("Key {} was not found".format(error))
+
+    """
+    Joins given list and appends a ':'
+    Expects a list
+    """
+    @staticmethod
+    def make_path_string(path_list):
+        return ' / '.join(path_list) + ":"
+
+
+    """
+    Prints the info and any children or parameters of the node.
+    """
+    @staticmethod
+    def print_help(node):
+        print(node.name + ":")
+        print("Info: " + node.command_info)
+        if node.parameter_list > 0:
```

Comment:
```suggestion
        if node.parameter_list:
```
previously this was checking that node.parameter_list has more then zero items,
now this is a comparison between a list and an int,
`TypeError: '>' not supported between instances of 'list' and 'int'`
to check wether a list is empty you should just use
```python
if list():
    pass
```

Reply:
Resolved in commit 
https://github.com/R2D2-2019/swarm_controller/pull/5/commits/ce29865273897da289128759a5e7650c1ea05dc1